cupyx/scipy/ndimage/filters.py

```diff
@@ -76,40 +164,105 @@ def _get_output(output, input, shape=None):
     return output
 
 
-def _correlate_or_convolve(input, weights, output, mode, cval, origin,
-                           convolution):
-    if input.dtype.kind == 'c':
-        raise TypeError('Complex type not supported.')
-    if not hasattr(origin, '__getitem__'):
-        origin = [origin, ] * input.ndim
+def _fix_sequence_arg(arg, ndim, name, conv=lambda x: x):
+    if hasattr(arg, '__iter__') and not isinstance(arg, str):
```

Comment:
we favor try/except AttributeError over `hasattr`

Reply:
@emcastillo Can you point me to some example code you want me to use?

Just curious why y'all prefer try/except over `hasattr`?
https://stackoverflow.com/questions/903130/hasattr-vs-try-except-block-to-deal-with-non-existent-attributes

Reply:
Sorry, yes, we use try/ except AttributeError because it is faster.
If you think that in this routine speed doesn't matter you can leave it as is 🙂

Reply:
I wonder if it would be cleaner and faster to check if `arg` is not a `str` and then just do `arg = iter(arg)`.

Reply:
So I tried swapping
```python
if hasattr(arg, '__iter__') and not isinstance(arg, str):
```
with 
```python
if not isinstance(arg, str):
    try:
        iter(arg)
    except AttributeError:
        return NotImplemented
```
Now most/all tests are failing. Are these not equivalent?

Reply:
The exception for an invalid iter with `iter` is
```
TypeError: 'object' object is not iterable
```
😅 this might be confusing